src/imagery/i.saocom/i.saocom.import/i.saocom.import.py

```diff
@@ -0,0 +1,271 @@
+# % description: Whether the data directory is zipped or not
+# %end
+# %option
+# % key: polarizations
+# % type: string
+# % required: yes
+# % multiple: yes
+# % answer: ['hh','hv','vh','vv']
+# % description: Polarizations to process (Default:['hh','hv','vh','vv'])
+# %end
+# %option
+# % key: multilook
+# % type: integer
+# % required: yes
+# % multiple: yes
+# % answer: 1,1
+# % description: Azimuth and range factors to apply (eg: [4,2]))
+# %end
+# %option G_OPT_R_OUTPUT
+# % key: basename
+# % description: Prefix for output raster map
+# % required: yes
+# %end
+
+
+import os
+import numpy as np
+import grass.script as gs
+from zipfile import ZipFile
+import rasterio
+import numpy as np
+from osgeo import gdal
+from affine import Affine
+import pandas as pd
+
+
+def apply_multilook(dataset, azLooks, rgLooks):
+    """
+    Esta función aplica un multilook dado a una imagen de entrada y actualiza la información de los metadatos
+    para la imagen de salida.
+
+    parámetros:
+    dataset: diccionario de datos con las llaves:
+        'array': contiene la matriz de datos
+        'metadata': contiene los metadatos con el formato dado por la librería rasterio
+    azLooks: número de looks a aplicar en dirección de azimuth
+    rgLooks: número de looks a aplicar en dirección de rango
+
+    retorna:
+    diccionario de datos con la estructura del dataset de entrada pero con el array resultante del
+    multilook y los metadatos actualizados
+    """
+
+
+    array = dataset["array"].copy()
+    metadata = dataset["metadata"].copy()
+
+    array = array[0]
+
+    cols = metadata["width"]
+    rows = metadata["height"]
+    new_rows = rows // azLooks
+    new_cols = cols // rgLooks
+    new_shape = (new_rows, azLooks, new_cols, rgLooks)
+
+    if rows % azLooks != 0 or cols % rgLooks != 0:
+        array = array[: new_rows * azLooks, : new_cols * rgLooks]
+
+    multilooked = array.reshape(new_shape).mean(-1).mean(1)
+    metadata["width"] = new_cols
+    metadata["height"] = new_rows
+    geoTs = metadata["transform"]
+    metadata["transform"] = Affine(
+        geoTs[0] * rgLooks, geoTs[1], geoTs[2], geoTs[3], geoTs[4] * azLooks, geoTs[5]
+    )
+
+    return {"array": np.expand_dims(multilooked, axis=0), "metadata": metadata}
+
+
+def read_bands_zip(data, pols):
+    with ZipFile(data, "r") as zfile:
+        file_list = [
+            img
+            for img in zfile.namelist()
+            if img.startswith("Data/") and not img.endswith(".xml")
+        ]
+    bands = {}
+    gcps = tuple()
+    dataset_pols = []
+    for l in file_list:
+        pol = l.split("-")[6]
+        dataset_pols.append(pol)
+        if pol in pols:
+            bands[pol] = {}
+            filename = f"/vsizip/{data}/{l}"
+            with rasterio.open(filename) as banda:
+                bands[pol]["array"] = banda.read()
+                bands[pol]["metadata"] = banda.meta
+                gcps = banda.get_gcps()
+    return bands, gcps, dataset_pols
```

Comment:
This is a lot of unnecessary code duplication. I would suggest using `zipfile.is_zipfile` to detect if it's a zip file automatically (and remove the associated module option) and then merging these 2 functions.